S2VX.Game/Story/Note/GameHoldNote.cs

```diff
@@ -19,25 +19,37 @@ public class GameHoldNote : HoldNote, IKeyBindingHandler<InputAction> {
         [Resolved]
         private PlayScreen PlayScreen { get; set; }
 
-        private enum HoldNoteState {
-            NotVisibleBefore,
+        public enum HoldNoteState {
```

Comment:
Wasn't it good practice to separate enums into their own files? Might need @maxrchung 's opinion on this.

Reply:
Yeah, I agree. I think we made an exception for private enums since those would never be used outside of that file, but public enums or classes should be put into a separate file.

Reply:
The test is the only place that uses HoldNoteState outside of this file. Otherwise it could be private. How should I proceed? Still put it in a separate file?

Reply:
Yeah, my preference is for all public enums and classes to be in a separate file.

Reply:
For more context, this is the rule that suggests not having nested public types: https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1034

I disabled this rule because NUnit tests can sometimes be written as one class per test, see [StoryMerge.Tests](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1034) that have examples of this.